internal/model/manifest.go

```diff
@@ -18,36 +17,72 @@ func (m ManifestName) String() string { return string(m) }
 // NOTE: If you modify Manifest, make sure to modify `Manifest.Equal` appropriately
 type Manifest struct {
 	// Properties for all manifests.
-	Name         ManifestName
-	tiltFilename string
+	Name          ManifestName
+	tiltFilename  string
+	dockerignores []Dockerignore
+	repos         []LocalGithubRepo
 
-	// TODO(maia): buildInfo
+	// Info needed to build an image. Can be StaticBuild, FastBuild... etc.
+	buildInfo buildInfo
 
 	// Info needed to deploy. Can be k8s yaml, docker compose, etc.
 	deployInfo deployInfo
 
-	// All docker stuff
-	cachePaths []string
-	dockerRef  reference.Named
-
 	// Properties for fast_build (builds that support
 	// iteration based on past artifacts)
 	BaseDockerfile string
 	Mounts         []Mount
 	Steps          []Step
 	Entrypoint     Cmd
+}
 
-	// From static_build. If StaticDockerfile is populated,
-	// we do not expect the iterative build fields to be populated.
-	StaticDockerfile string
-	StaticBuildPath  string // the absolute path to the files
-	StaticBuildArgs  DockerBuildArgs
+type DockerBuildArgs map[string]string
 
-	dockerignores []Dockerignore
-	repos         []LocalGithubRepo
+func (m Manifest) WithBuildInfo(info buildInfo) Manifest {
+	m.buildInfo = info
+	return m
 }
 
-type DockerBuildArgs map[string]string
+func (m Manifest) DockerInfo() DockerInfo {
+	switch info := m.buildInfo.(type) {
```

Comment:
would this be simpler as a plain type cast? not sure what the extra machinery of a `switch` is getting you

Reply:
ha fair.